### PR TITLE
[YUNIKORN-185] K8shim logs cleanup

### PR DESCRIPTION
**Pull request by yangwwei**
This PR does a few optimizations for the logs
- remove the predicate errors from the logging, we can leverage the event system, no need to log in the file
- for state transition logs, remove dup records
- leverage zap post-state hook to log event transitions, added unit tests

### CODE REVIEW

**Comment by yangwwei** [comment]
I have to remove the unit tests for those event loggings
because of the race condition.

File: pkg/cache/application.go [hunk 1]
```diff
@@ -355,7 +348,10 @@ func (app *Application) handleCompleteApplicationEvent(event *fsm.Event) {
 
 func (app *Application) enterState(event *fsm.Event) {
 	log.Logger.Info("app state",
+		zap.String("currentState", event.Src),
+		zap.String("targetState", event.Dst),
+		zap.String("event", event.Event),
 		zap.String("appID", app.applicationID),
 		zap.String("queue", app.queue),
-		zap.String("state", app.GetApplicationState()))
+		zap.String("currentAppState", app.GetApplicationState()))
```
Comment (wilfred-s) [inline pkg/cache/application.go:356]: This is a double up: we have current and target state already (currentAppState)
Can we also bring this inline with the way this is done in the core? Log at a debug level and with minimal info and the same order: appID, source, destination, and event.

File: pkg/cache/application_test.go [hunk 2]
```diff
@@ -23,6 +23,7 @@ import (
 	"time"
 
 	"gotest.tools/assert"
+
```
Comment (wilfred-s) [inline pkg/cache/application_test.go:26]: spurious new line

File: pkg/cache/node.go [hunk 3]
```diff
@@ -245,3 +239,12 @@ func (n *SchedulerNode) canHandle(ev events.SchedulerNodeEvent) bool {
...
 	return n.fsm.Can(string(ev.GetEvent()))
 }
+
+func (n *SchedulerNode) enterState(event *fsm.Event) {
+	log.Logger.Info("node state",
+		zap.String("currentState", event.Src),
+		zap.String("targetState", event.Dst),
+		zap.String("event", event.Event),
+		zap.String("nodeName", n.name),
+		zap.String("currentNodeState", n.getNodeState()))
```
Comment (wilfred-s) [inline pkg/cache/node.go:249]: This is a double up: we have current and target state already (currentAppState).
Can we also bring this inline with the way this is done in the core? Log at a debug level and with same info order: node, source, destination, and event.

File: pkg/cache/task.go [hunk 4]
```diff
@@ -415,8 +408,11 @@ func (task *Task) sanityCheckBeforeScheduling() error {
...
 func (task *Task) enterState(event *fsm.Event) {
 	log.Logger.Info("task state",
+		zap.String("currentState", event.Src),
+		zap.String("targetState", event.Dst),
+		zap.String("event", event.Event),
 		zap.String("appID", task.applicationID),
 		zap.String("taskID", task.taskID),
 		zap.String("taskAlias", task.alias),
-		zap.String("state", task.GetTaskState()))
+		zap.String("currentTaskState", task.GetTaskState()))
```
Comment (wilfred-s) [inline pkg/cache/task.go:417]: This is a double up: we have current and target state already (currentAppState).
Log at a debug level and with same info order: taskID, source, destination, and event. If we really need the appID to track it as the task ID is not unique then we should add that back.

File: pkg/cache/task_test.go [hunk 5]
```diff
@@ -23,13 +23,13 @@ import (
 
 	"gotest.tools/assert"
```
Comment (wilfred-s) [inline pkg/cache/task_test.go:25]: remove the empty line

File: pkg/shim/scheduler.go [hunk 6]
```diff
@@ -282,6 +268,14 @@ func (ss *KubernetesShim) run() {
 	ss.apiFactory.Start()
 }
 
+func (ss *KubernetesShim) enterState(event *fsm.Event) {
+	log.Logger.Info("scheduler shim state",
+		zap.String("currentState", event.Src),
+		zap.String("targetState", event.Dst),
+		zap.String("event", event.Event),
+		zap.String("currentSchedulerState", ss.GetSchedulerState()))
+}
```
Comment (wilfred-s) [inline pkg/shim/scheduler.go:277]: This is a double up: we have current and target state already (currentAppState).
Log at a debug level and with same info order: shim identification, source, destination and event.

File: pkg/cache/application.go (hunk 1 above)
Comment (yangwwei) [reply pkg/cache/application.go:356]: Make sense. Align with how the state transition is logged in core.

File: pkg/cache/task.go (hunk 4 above)
Comment (yangwwei) [reply pkg/cache/task.go:417]: We need appID, taskID and alias for tracking purpose, revised to

```
	log.Logger.Debug("shim task state transition",
		zap.String("app", task.applicationID),
		zap.String("task", task.taskID),
		zap.String("taskAlias", task.alias),
		zap.String("source", event.Src),
		zap.String("destination", event.Dst),
		zap.String("event", event.Event))
```

File: pkg/cache/application_test.go (hunk 2 above)
Comment (yangwwei) [reply pkg/cache/application_test.go:26]: Removed

File: pkg/cache/task_test.go (hunk 5 above)
Comment (yangwwei) [reply pkg/cache/task_test.go:25]: removed

**Pull request closed without merging**